karapace/schema_reader.py

```diff
@@ -140,21 +163,21 @@ def parse(schema_type: SchemaType, schema_str: str):  # pylint: disable=inconsis
         raise InvalidSchema(f"Unknown parser {schema_type} for {schema_str}")
 
     def to_json(self):
-        if isinstance(self.schema, Draft7Validator):
+        if self.schema_type is SchemaType.AVRO:
```

Comment:
These are changed so no parsing to target schema type is done. The original schema string is used for creating the JSON repr.
This function and also `str` and `repr` are called when constructing f-strings for logging.